src/sonic-yang-models/yang-models/sonic-scheduler.yang

```diff
@@ -0,0 +1,149 @@
+module sonic-scheduler {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-scheduler";
+
+    prefix sch;
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SCHEDULER yang Module for SONiC OS";
+
+    revision 2021-04-01 {
+        description
+            "Initial revision";
+    }
+
+    container sonic-scheduler {
+
+        container SCHEDULER {
+
+            description "QUEUE part of config_db.json";
+
+            list SCHEDULER_LIST {
+
+                key "name";
+
+                leaf name {
+
+                    type string {
+                        pattern "[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})|[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})([@]{1})([0-9]{1,3})";
+                        length 1..36;
+                    }
+                    description "Scheduler name";
+                }
+
+                leaf type {
+                    type enumeration {
+                        enum DWRR {
+                            description "Deficit Weighted Round-Robin Scheduling";
+                        }
+                        enum WRR {
+                            description "Weighted Round-Robin Scheduling";
+                        }
+                        enum STRICT {
+                            description "Strict Scheduling";
+                        }
+                    }
+                    default WRR;
+                    description "Scheduling algorithm type";
+                }
+
+                leaf weight {
+                    type uint8 {
+                        range "1..100";
+                    }
+                    default 1;
+                    description "Scheduling algorithm weight";
+                }
+
+                leaf priority {
+                    type uint8 {
+                        range "0..9";
+                    }
+                    description "Scheduler priority";
+                }
+
+                leaf meter_type {
+                    type enumeration {
+                        enum packets {
+                            description "Metering is based on packets";
+                        }
+                        enum bytes {
+                            description "Metering is based on bytes";
+                        }
+                    }
+                    default bytes;
+                    description "meter type";
+                }
+
+                leaf cir {
+                    type uint64;
+                    default 0;
+                    description
+                        "Committed information rate for the dual-rate token
```

Comment:
CIR is well know industry term. Mentioning the units is good enough in my opinion.

"CIR in bps for meter type 'bps' and pps for meter type 'pps'"

Reply:
@smaheshm - do you see any issue if we explain details in description?

Reply:
No issues, leave it up to you.